File: ios/MidTrim/MidTrim/Domain/UseCases/RenameProjectUseCase.swift
import Foundation

struct RenameProjectUseCase {
    private let repository: ProjectRepositoryProtocol

    init(repository: ProjectRepositoryProtocol) {
        self.repository = repository
    }

    func execute(projectID: UUID, newName: String) async throws {
        let trimmed = newName.trimmingCharacters(in: .whitespaces)
        guard !trimmed.isEmpty else {
            throw ProjectError.emptyName
        }
        try await repository.rename(by: projectID, to: trimmed)
    }
}

File: android/app/src/main/java/com/izzatismail/midtrim/domain/usecase/RenameProjectUseCase.kt
package com.izzatismail.midtrim.domain.usecase

import com.izzatismail.midtrim.domain.error.ProjectError
import com.izzatismail.midtrim.domain.repository.ProjectRepository

class RenameProjectUseCase(
    private val repository: ProjectRepository
) {
    suspend fun execute(projectId: String, newName: String) {
        if (newName.isBlank()) {
            throw ProjectError.EmptyName
        }
        repository.rename(projectId, newName.trim())
    }
}